drivers/adc/ad4858/ad4858.c

```diff
@@ -481,8 +494,13 @@ int ad4858_set_chn_offset(struct ad4858_dev *dev, uint8_t chn, uint32_t offset)
 	if (!dev || (chn >= AD4858_NUM_CHANNELS))
 		return -EINVAL;
 
-	/* Offset value is spanned over bits [24:4] */
-	ret = ad4858_reg_write(dev, AD4858_REG_CH_OFFSET(chn), (offset << 4));
+	if (dev->prod_id == AD4855_PROD_ID_L)
+		/* Offset value is spanned over bits [23:8] */
```

Comment:
this is a bug

```
if (condition)
    // comment
    do1();
else
    // comment
    do2();
```

it executes like this:
```
if (condition)
    // comment
do1();  // Always executes
else
    // comment
do2();  // Always executes
```

And this is not what you wanted. That's why MISRA enforces {} for all if/else statements even if they enclose oneliners. Put brackets and you'll be fine.

Reply:
also, the compiler gave you warnings that you ignored, i'm pretty sure you'd get a warnings that the if/else does nothing

Reply:
It worked as expected on my end
I've added the brackets to make it unambiguous